File: 575-distribute-candies/distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        unordered_map<int,int>k;
        for(auto c:candyType){
            k[c]++;
        }
        int f = (candyType.size())/2;
        int y = k.size();
        if(f<=y){
            return f;
        }
        else{
            return y;
        }
    }
};

File: 575-distribute-candies/distribute-candies.py
from collections import Counter
class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        k = Counter(candyType)
        m=len(k)
        if(len(candyType)/2 <=m):
            return int(len(candyType)/2)
        else:
            return m